server/models/UserModel/index.js

```diff
@@ -58,6 +67,34 @@ const UserModel = {
 
     return value
   },
+  hasAdmin: async (db) => {
+    const userCt = await db
+      .collection(collections.users)
+      .countDocuments({ role: "admin" })
+    return userCt !== 0
+  },
+  createFirstAdmin: async (db) => {
+    if (await UserModel.hasAdmin(app.locals.appDb)) {
+      return
+    } 
+
+    const reset_key = crypto.randomBytes(32).toString('hex')
+
+    await UserModel.create(db, {
+      uid: 'admin',
+      password: null,
+      role: 'admin',
+      force_reset_pw: true,
+      reset_key,
+    })
```

Comment:
do we need to set access?

Reply:
Nope! We'll get access on update, including when the password is set for the new account, so it's not needed here.